Service/src/main/java/Repository/ThreadModelRep/ThreadRepositoryImplInMemory.java

```diff
@@ -0,0 +1,249 @@
+            result.put((Student)student, ((Student) student).getListOfMark());
+        }
+        return result;
+    }
+
+    @Override
+    public List<Mark> getMarkListbyTheam(Theams theam, int studentId) {
+        Student student = (Student) DataBaseInf.studentHashMap.get(studentId);
+        return student.getListOfMark().get(theam);
+    }
+
+    @Override
+    public HashMap<Integer, Mark> getMarkIDListbyTheam(Theams theam, int studentId) {
+        HashMap<Integer, Mark> result = new HashMap<>();
+        for (Mark mark:
+             getMarkListbyTheam(theam, studentId)) {
+            result.put(mark.getId(), mark);
+        }
+        return  result;
+    }
+
+    @Override
+    public Theams theamById(Integer id) {
+        return DataBaseInf.theamsHashMap.get(id);
+    }
+
+    @Override
+    public Set<Theams> theamFromGroup(Integer groupId) {
+        Group group = DataBaseInf.groupHashMap.get(groupId);
+        return group.getTheamsSet();
+    }
+
+    @Override
+    public HashMap<Integer, UserImpl> studentsFromGroup(int groupId) {
+        Group group = DataBaseInf.groupHashMap.get(groupId);
+        return (HashMap<Integer, UserImpl>) group.getStudentMap();
+    }
+
+    @Override
+    public void addTheam(String theam) {
+    Theams theams = new Theams()
+            .withId(IdFactory.idBuilder())
+            .withValue(theam);
+    DataBaseInf.theamsHashMap.put(theams.getId(), theams);
+    }
+
+    @Override
+    public void addGroup(List<UserImpl> studentList, List<Integer> theamsIdList, Integer trainerId) {
+    Group group = new Group();
+    Set <Theams> theams =  new HashSet<>();
+    HashMap <Integer, UserImpl> students = new HashMap<>();
+        for (UserImpl user:
+             studentList) {
+            Student student = (Student)user;
+            for (Integer theamId:
+                 theamsIdList) {
+                Theams tempTheam = theamById(theamId);
+                student.addTheam(tempTheam);
+            }
+            students.put(student.getId(), student);
+        }
+        for (Integer theamId:
+             theamsIdList) {
+            theams.add(theamById(theamId));
+        }
+    group.withId(IdFactory.idBuilder())
+            .withName("Groups_" + group.getId() )
+            .withStudents(students)
+            .withTrainer(RepositoryFactory.getRepository().getUserById(trainerId))
+            .withTheam(theams);
+        log.info("Group was create = {}", "Group name:" + group.getName() +
+                " group student: " +group.getStudentMap().values()  +" " + group.getTheamsSet().toString());
+        DataBaseInf.groupHashMap.put(group.getId(), group);
+    }
+
+    @Override
+    public HashMap<Integer, Theams> freeTheams() {
+        HashMap <Integer, Theams> result;
+        if (DataBaseInf.groupHashMap.values().isEmpty()) {
+            return (HashMap<Integer, Theams>) DataBaseInf.theamsHashMap;
+        }
+            ArrayList<Theams> busyTheams = new ArrayList<>();
```

Comment:
Всегда старайся использовать общий класс или интерфейс
```suggestion
            List<Theams> busyTheams = new ArrayList<>();
```

Reply:
Исправил.